assets/react-hooks/use-select-with-debounce.test.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import useSelectWithDebounce from './use-select-with-debounce';
+
+jest.mock( '@wordpress/data' );
+
+describe( 'useSelectWithDebounce', () => {
+	it( 'Should change the deps to call useSelect after timer only', () => {
+		jest.useFakeTimers();
+		const mapSelect = () => {};
+		const timer = 1000;
+		const mockFn = jest.fn();
+		useSelect.mockImplementation( mockFn );
+
+		const TestComponent = ( { deps } ) => {
+			useSelectWithDebounce( mapSelect, deps, timer );
+			return <div />;
+		};
+
+		const deps1 = [ 1 ];
+		const deps2 = [ 2 ];
+
+		const { rerender } = render( <TestComponent deps={ deps1 } /> );
+		expect( mockFn ).toBeCalledWith( mapSelect, deps1 );
+
+		rerender( <TestComponent deps={ deps2 } /> );
+		expect( mockFn ).toBeCalledWith( mapSelect, deps1 );
+
+		// TODO: Complete test running debounce to make sure it was called after the time.
+		// jest.runAllTimers();
+		// expect( mockFn ).toBeCalledWith( mapSelect, deps2 );
```

Comment:
Do you plan to do this before merging?

Reply:
Unfortunately, I don't plan to add it now. We're having some problems with dependencies in order to make it work. `runAllTimers` is starting an infinite loop in the `debounce`. There are some issues about that on the Internet and it was supposed to be fixed, so at some point, we need to dig into the dependencies to see what's happening.

Reply:
Just adding some links here for future reference:

https://github.com/facebook/jest/issues/3465
https://github.com/lodash/lodash/issues/2893
https://github.com/facebook/jest/pull/7776